src/System.Console/src/System/TermInfo.cs

```diff
@@ -119,10 +119,20 @@ private Database(string term, byte[] data)
                 _term = term;
                 _data = data;
 
+                const int magicLegacyNumber = 0x11A; // magic number octal 0432 for legacy ncurses terminfo
+                const int magic32BitNumber = 0x21E; // magic number octal 01036 for new ncruses terminfo
+                bool reasAs32Bit = false; // default to legacy format
+
+                short magic = ReadInt16(data, 0);
+
                 // See "man term" for the file format.
-                if (ReadInt16(data, 0) != 0x11A) // magic number octal 0432
+                if (magic == magic32BitNumber)
                 {
-                    throw new InvalidOperationException(SR.IO_TermInfoInvalid);
+                    reasAs32Bit = true; // Using new ncurses format
+                }
+                else if (magic != magicLegacyNumber)
+                {
+                    throw new InvalidOperationException(SR.IO_TermInfoInvalid); // Did not recognize legacy or new magic numbers
```

Comment:
Suggestion: include the magic number in the text. That way next time this changes, the error will include the magic number and we'll know it's a format change and not "the term info database is corrupt" which we throw in 7 other places.